File: csharp/LogicAndTrick.WikiCodeParser/Elements/MdQuoteElement.cs
using System;
using System.Collections.Generic;
using LogicAndTrick.WikiCodeParser.Nodes;

namespace LogicAndTrick.WikiCodeParser.Elements
{
    public class MdQuoteElement : Element
    {
        public override bool Matches(Lines lines)
        {
            var value = lines.Value();
            return value.Length > 0 && value[0] == '>';
        }

        public override INode Consume(Parser parser, ParseData data, Lines lines, string scope)
        {
            var value = lines.Value();
            var arr = new List<string>
            {
                value.Substring(1).Trim()
            };
            while (lines.Next())
            {
                value = lines.Value().Trim();
                if (value.Length == 0 || value[0] != '>')
                {
                    lines.Back();
                    break;
                }
                arr.Add(value.Substring(1).Trim());
            }

            var text = String.Join("\n", arr);
            return new HtmlNode("<blockquote>", parser.ParseElements(data, text, scope), "</blockquote>");
        }
    }
}

File: php/src/Elements/MdQuoteElement.php
<?php

namespace LogicAndTrick\WikiCodeParser\Elements;

use LogicAndTrick\WikiCodeParser\Lines;
use LogicAndTrick\WikiCodeParser\Nodes\HtmlNode;
use LogicAndTrick\WikiCodeParser\Nodes\INode;
use LogicAndTrick\WikiCodeParser\ParseData;
use LogicAndTrick\WikiCodeParser\Parser;

class MdQuoteElement extends Element
{
    public function Matches(Lines $lines): bool
    {
        $value = $lines->Value();
        return strlen($value) > 0 && str_starts_with($value, '>');
    }

    public function Consume(Parser $parser, ParseData $data, Lines $lines, string $scope): ?INode
    {
        $value = $lines->Value();
        $arr = [trim(substr($value, 1))];
        while ($lines->Next()) {
            $value = trim($lines->Value());
            if (strlen($value) == 0 || $value[0] != '>') {
                $lines->Back();
                break;
            }
            $arr[] = trim(substr($value, 1));
        }

        $text = implode("\n", $arr);
        return new HtmlNode('<blockquote>', $parser->ParseElements($data, $text, $scope), '</blockquote>');
    }
}